source/connection-monitoring-and-pooling/tests/cmap-format/pool-checkout-error-concurrent.yml

```diff
@@ -0,0 +1,47 @@
+version: 1
+style: integration
+description: must properly handle multiple concurrent network errors on check out
+runOn:
+  -
+    # required for appName in fail point
+    minServerVersion: "4.9.0"
+failPoint:
+  configureFailPoint: failCommand
+  # high amount to ensure not interfered with by monitor checks.
+  mode: { times: 50 }
+  data:
+    failCommands: ["isMaster","hello"]
+    closeConnection: true
+    appName: &appName "poolCheckOutErrorTest"
+poolOptions:
+  appName: *appName
+operations:
+  - name: ready
+  - name: start
+    target: thread1
+  - name: start
+    target: thread2
+  - name: start
+    target: thread3
+  - name: checkOut
+    thread: thread1
+  - name: checkOut
+    thread: thread2
+  - name: checkOut
+    thread: thread3
+  - name: waitForEvent
+    event: ConnectionCheckOutFailed
+    count: 3
+  - name: ready
+events:
+  - type: ConnectionPoolReady
```

Comment:
The cmap-format tests are unit tests of an individual pool, so the events in each one should only be associated with that pool. If a driver is testing this via an actual client which makes pools per server, I think it would probably be best if the test runner for that driver just filtered out any events observed from other pools.